### ISSUE TITLE
Peer review Lab 2

### ISSUE THREAD

**Issue by francescoscalera99** [rogierz/ComputationalIntelligence2022#4]
Hi Ruggero!
First of all, I want to tell you that your code is written is a very good way, readable and clear, thanks to the way you've implemented the Individual class.
Since the fact you have numpy array and some strategies to decrease computation time (like binary representation to check the covered numbers, that I appreciate a lot), I notice that the tournament selection for selecting parents is too slow, and I think because you involve a large number of individuals. To speed up this phase, you can try to do the tournament selection between a smaller number of individual chosen randomly (in my solution I have 2). 
In conclusion, in order to have an heterogeneous population, and thus to have a more important effect when you apply genetic operators, I suggest you to remove duplicates from population and the offspring at every generation, you may have a better result.
